ui/app/store/actions.js

```diff
@@ -2472,13 +2472,16 @@ export function clearPermissions() {
  * @param {any} [value] - The value required to confirm a pending approval
  */
 export function resolvePendingApproval(id, value) {
-  return async (dispatch) => {
+  return async () => {
     await promisifiedBackground.resolvePendingApproval(id, value);
     // Before closing the current window, check if any additional confirmations
     // are added as a result of this confirmation being accepted
     const { pendingApprovals } = await promisifiedBackground.getState();
```

Comment:
Perhaps you could use `forceUpdateMetamaskState` here instead, and preserve the `closeCurrentNotificationWindow` call? Generally that should be preferred to getting update state for this specific action, to avoid the UI getting into an inconsistent state (has happened here).

Reply:
I did this, but it was still presenting this way... Probably because there's a race condition there where if the clearing of that pending transaction confirmation doesn't happen before the check the notification window isn't closed. I realized though that something wasn't working right in the confirmation screen because it should be re-routing to the homepage when there are no approvals pending. I was for some reason excluding the notification window from this logic, presumably because it should close, but closing is also handled elsewhere when examining the state. 

I think its still a better call to forceUpdate the state here so i made that change too.